[sc-59960] Migrate to pnpm

hieptd-zinza opened the pull request:
Updated `action-hasura-change-summary` to use `pnpm`

hieptd-zinza committed:
[sc-59960] update Fieldguide action hasura change summary

hieptd-zinza committed:
[sc-59960] update ci.yml

hieptd-zinza committed:
Build project

twhitbeck commented on package.json:
```diff
@@ -6,6 +6,7 @@
   "engines": {
     "node": ">=20.x <21"
   },
+  "packageManager": "pnpm@10.12.1",
```
Let's go with `10.12.4`

namoscato committed:
pnpm import

Reset current state:
$ git checkout main package.json
$ git checkout main package-lock.json
$ git checkout main src/load/yaml.ts
$ rm pnpm-lock.yaml
$ rm pnpm-workspace.yaml

Update package.json, setting:
"packageManager": "pnpm@10.12.4"

Import from package-lock.json
$ corepack enable
$ pnpm import
$ pnpm format
$ pnpm all
$ rm package-lock.json

namoscato committed:
Add @types/prettier

pnpm add -D @types/prettier@^2

namoscato committed:
Add @octokit/types

pnpm add -D @octokit/types

namoscato reviewed (commented):
@twhitbeck, do you mind giving this a review given I reworked most of this?

🚩 @Fieldguide/zinza, please be sure to follow [the runbook](https://www.notion.so/fieldguide/Migrate-to-pnpm-21eadc3e6ca380ab8221ce789c0e9713) closely in the future. Notably, it seems that [`pnpm import`](https://pnpm.io/cli/import) was not leveraged, inadvertently upgrading a number of packages, requiring unnecessary source code changes.

b68fe251689b2c013f2121e8af32c4bd51182d6d re-works most of this PR (with steps detailed in the commit description), and the followup commits install some missing dev dependencies due to pnpm strict transitive dependency behavior.
